File: legacy/expo-rn/src/utils/env.ts
export const PUBLIC_ENV_ERROR_PREFIX = 'Missing required Expo public env';

export function resolveRequiredPublicEnv(
  env: Record<string, string | undefined>,
  key: string
): string {
  const value = env[key]?.trim();
  if (!value) {
    throw new Error(`${PUBLIC_ENV_ERROR_PREFIX}: ${key}`);
  }
  return value;
}

File: apps/android-native/app/src/main/java/com/wxxtae/pws/nativepreview/domain/Env.kt
package com.wxxtae.pws.nativepreview.domain

const val PublicEnvErrorPrefix: String = "Missing required Expo public env"

object PwsEnv {
    fun resolveRequiredPublicEnv(env: Map<String, String?>, key: String): String {
        val value = env[key]?.trim().orEmpty()
        if (value.isBlank()) throw IllegalStateException("$PublicEnvErrorPrefix: $key")
        return value
    }
}
